src/v/cloud_storage/types.cc

```diff
@@ -262,17 +267,53 @@ size_t anomalies::count_segment_meta_anomaly_type(anomaly_type type) const {
     return static_cast<size_t>(count);
 }
 
+/// Returns number of discarded elements
+template<class T, size_t quota = max_number_of_manifest_anomalies>
+size_t insert_with_size_limit(
+  absl::node_hash_set<T>& self, const absl::node_hash_set<T>& other) {
```

Comment:
nit: I found `self` to be a bit confusing, since it's easy to read this and think "I'm reading the a class method, oh wait no I'm not, what's going on?". Maybe consider something like `dest` and `to_add` or something

Reply:
done